src/main/java/com/lambdatest/jenkins/freestyle/report/AppAutomationReportBuildAction.java

```diff
@@ -0,0 +1,135 @@
+            InputStream is = buildUrlConnection.getInputStream();
+            InputStreamReader isr = new InputStreamReader(is);
+            int numCharsReadForBuild;
+            char[] charArrayForBuild = new char[1024];
+            StringBuffer sb = new StringBuffer();
+            while ((numCharsReadForBuild = isr.read(charArrayForBuild)) > 0) {
+                sb.append(charArrayForBuild, 0, numCharsReadForBuild);
+            }
+            String buildResult = sb.toString();
+            try {
+                ObjectMapper mapper = new ObjectMapper();
+                JsonNode rootNode = mapper.readTree(buildResult);
+                ArrayNode dataNode = (ArrayNode) rootNode.get("data");
+                Iterator<JsonNode> dataIterator = dataNode.elements();
+                while (dataIterator.hasNext()) {
+                    JsonNode buildDetailNode = dataIterator.next();
+                    if (buildDetailNode.get("name").toString().replaceAll("\"", "").equals(buildName)) {
+                        String build_id = buildDetailNode.get("build_id").toString();
+
+                        lambdaTestBuildDeviceUrl = "https://appautomation.lambdatest.com/test?buildid=" + build_id;
+
+                        logger.info("lambdaTestBuildDeviceUrl : " + lambdaTestBuildDeviceUrl);
+                        URL sessionUrl = new URL(Constant.AppAutomationReport.SESSION_INFO_URL + "?limit=100&build_id=" + build_id);
+                        URLConnection sessionUrlConnection = sessionUrl.openConnection();
+                        sessionUrlConnection.setRequestProperty("Authorization", "Basic " + authStringEnc);
+                        is = sessionUrlConnection.getInputStream();
+                        isr = new InputStreamReader(is);
+                        int numCharsReadForSession;
+                        char[] charArrayForSession = new char[1024];
+                        sb = new StringBuffer();
+                        while ((numCharsReadForSession = isr.read(charArrayForSession)) > 0) {
+                            sb.append(charArrayForSession, 0, numCharsReadForSession);
+                        }
+                        String sessionResult = sb.toString();
+
+                        JsonNode sessionRootNode = mapper.readTree(sessionResult);
+                        ArrayNode sessionDataNode = (ArrayNode) sessionRootNode.get("data");
+                        Iterator<JsonNode> sessionDataIterator = sessionDataNode.elements();
+
+                        JSONArray array = new JSONArray();
+                        while (sessionDataIterator.hasNext()) {
+                            JsonNode sessionDetailNode = sessionDataIterator.next();
+                            logger.info("sessiondetailnode : " + sessionDataNode.toString());
+                            String testUrl = "https://appautomation.lambdatest.com/test?testID=" + sessionDetailNode.get("test_id").toString().replaceAll("\"", "") + "&buildid=" + build_id;
```

Comment:
Please use constant value `APP_AUTOMATION_APP_URL` here to make URL

Reply:
Done